File: Python/Block.py
'''File name: Block.py
Created by Joshua Brock on 9/23/2018'''


class Block:

    def __init__(self, statements):
        '''Precondition: statements is valid
        Postcondition: default constructor - creates a new block with list of statements'''
        assert (isinstance(statements, list))
        self.statements = []
        self.statements.extend(statements)

    def execute(self):
        '''Postcondition: executes every statement in the list in the order they were added'''
        for x in self.statements:
            x.execute()

File: Java/Block.java
import java.util.ArrayList;

public class Block {
    private ArrayList<Statemnt> statements;

    public Block(ArrayList<Statemnt> statements)
    {
        //Creates a new program containing 1 block.
        this.statements = statements;
    }

    public void execute()
    {
        //Postcondition: Executes all statements in the block.
        for(int i = 0; i < statements.size(); i++)
            statements.get(i).execute();
    }
}
